portal-web/docroot/html/common/themes/top_head.jsp

```diff
@@ -20,7 +20,7 @@
 
 <liferay-util:dynamic-include key="/html/common/themes/top_head.jsp#pre" />
 
-<link href="<%= BrowserSnifferUtil.isIe(request) ? StringPool.BLANK : themeDisplay.getPathThemeImages() %>/<%= PropsValues.THEME_SHORTCUT_ICON %>" rel="icon" />
```

Comment:
hey @fortunatomaldonado, @jonmak08 Sorry for the delay, I just came back from vacation. But I just want to leave my worries about reversing this, we did this so that Favicon appears in the navigations with Senna in IE 11, the solution is always to point to the root... We have a [servlet](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-theme/frontend-theme-favicon-servlet/src/main/java/com/liferay/frontend/theme/favicon/servlet/internal/FaviconServlet.java) that redirects to the favicon of the theme... I do not know if this will correct the problem, since in navigation with senna the favicon will disappear. I think we're trading for each other...

The favicon customized will appear in the first navigations and after that IE saves the icon of the favicon in a local folder to be able to reuse...

Reply:
Hello @matuzalemsteles, not a problem. Thank you for the detailed explanation. It seems that the servlet is always pointing to the classic-theme favicon even if a custom theme is being used causing the issue in LPS-88702. Is the servlet suppose to always point to the classic theme?

Reply:
From what I know, the servlet fetches the favicon from the theme according to the request.

Reply:
Oh okay. So as of now it is always point to classic theme and is not recognizing any custom theme being used.

Reply:
hey @fortunatomaldonado, I don't know if I understand what you mean.

Reply:
Hello @matuzalemsteles, I apologize for not being clear. Currently in IE11, the favicon being displayed with always be the default liferay favicon. The issue lies in having a custom theme with a different favicon icon. The default liferay favicon will be displayed even if the user has the custom theme in IE11. Reverting the fix, will result in the favicon disappearing which is the documented issue by Microsoft found here: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/9383036/
I am not sure if we can properly resolve this issue in IE11, but I believe losing the favicon would be more appropriate than displaying an incorrect favicon for a custom theme.
Let me know your thoughts on this. Thank you.

Reply:
Thanks for the clarifications, I'll take a look at this tomorrow and check if the servlet is actually doing what is expected, I'm guessing it's more of an IE cache problem, it changes the favicon only when its name changes.

Reply:
Thank you, truly appreciate it!

Reply:
Side Note: With the current implementation, there is a limited workaround for IE11 favicon. If the customer replaces the classic-theme favicon with their desired favicon, it will be displayed in IE11. This will only be useful if the customer has one favicon that they would like to use.

Reply:
hey @fortunatomaldonado this is an option... I've been testing if the servlet redirection was being done correctly according to the theme, I was able to test this successfully, can you test it? Adding the theme and testing manually, `localhost:8080/favicon.ico` it should go to the address of the theme where the favicon is.

Reply:
Hello @matuzalemsteles, yes I was able to test this and it does get redirected to the address of the theme that is being set as "default theme." in instance settings. This causes an issue in ie11 for pages that have a different theme set in the look and feel configurations. The default theme favicon will override the page's theme favicon. In other browsers, the page's theme favicon will be displayed.

Reply:
hey @fortunatomaldonado, yes this is a problem, I really do not know if the best case would be to remove this just for that case and all others will be affected... hey @jbalsas do you have any suggestions about this?

Reply:
Hey @fortunatomaldonado, have you checked the logic in https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-theme/frontend-theme-favicon-servlet/src/main/java/com/liferay/frontend/theme/favicon/servlet/internal/FaviconServlet.java#L51-L69? Maybe we need to reverse the logic or add some additional checks to cover the case you mention?

Reply:
In any case, I agree that reverting is not the best course of action here. Let's try to figure out why we're not redirecting to the right path.

Reply:
Hello @jbalsas and @matuzalemsteles,
I have looked at it briefly and it seems like it looks for the ThemeDisplay but it always returns null. It will then call the LayoutSet and that is where the default theme is being used. I believe if we can get the ThemeDisplay we will be able to get the current theme's favicon, unless I am misunderstanding how this is working.

Reply:
Hello @jbalsas and @matuzalemsteles,
I have tried getting the ThemeDisplay, Theme, Layout, but keep getting null for these requests. I am not sure how to go about this and any advice is much appreciated.
Thanks again.

Reply:
Hey @fortunatomaldonado I'm not sure how to handle this part, maybe @jbalsas can help you better about it. So I will return the LPS to you so we can see how to proceed with this. All right?

Reply:
Hello @matuzalemsteles, yeah that sounds good. Thank you for your help.
@jbalsas, any help on this is truly appreciated.

Reply:
Hello @jbalsas, I was testing this fix and I am not sure how to get the parameter in the servlet. From my understanding, after appending the themeId in the url it should be added to the url in the servlet. Let me know if this is correct or not. 
I also found that in IE another query `?q=NUMBERS` gets appended to the url after changing pages. This is another issue since I believe we cannot have two `?` in a url.
I appreciate all the help you have provided so far and now have a better understanding on how this is working for IE11.

Reply:
hey @fortunatomaldonado, We are forcing the addition of the query `?q=` in order to force a favicon request when the links are different, this helps to not cache the most current browsers. I think we can do with adding more than one in the query, you [can take a look here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/screen/EventScreen.es.js#L306-L318).

Reply:
Hello @matuzalemsteles, thank you for the information it will come in handy. 
I was able to find that the link was being rewritten here. https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/WEB-INF/urlrewrite.xml#L9-L12
I will continue tomorrow with this and update on the progress.
Thank you again.

Reply:
Hello,
I have partially implemented Chema's suggestions and I have been having trouble with the addition of the query. I am trying to replace the addition query with an & but for one reason or another it is not taking. This could be an issue on my side. I am currently trying to find another way around this using the urlrewrite and adding the theme name in the url. I will continue with this and let you know my progress.
Let me know if there are any questions or comments about this.
Thank you all again.